sqlglot/parser.py

```diff
@@ -7083,35 +7093,109 @@ def _parse_alter(self) -> exp.Alter | exp.Command:
+        while self._match_texts(self.ANALYZE_STYLES):
+            options.append(self._prev.text.upper())
 
-        if self._match(TokenType.TABLE):
+        kind = None
+        this: t.Optional[exp.Expression] = None
+        partition: t.Optional[exp.Expression] = None
+
+        if self._match(TokenType.TABLE) or self._match(TokenType.INDEX):
+            kind = self._prev.text.upper()
             this = self._parse_table_parts()
-            partition = self._parse_partition()
-        elif self._match_texts("TABLES"):
+        elif self._match_text_seq("TABLES"):
+            kind = self._prev.text.upper()
             this = (
-                self._parse_table(is_db_reference=True)
+                self._parse_table(schema=True, is_db_reference=True)
                 if self._match_set((TokenType.FROM, TokenType.IN))
                 else None
             )
         else:
-            return self._parse_as_command(start)
+            # Empty kind  https://prestodb.io/docs/current/sql/analyze.html
+            this = self._parse_table_parts()
 
-        if self._match_text_seq("COMPUTE", "STATISTICS"):
-            return self.expression(
-                exp.Analyze,
-                kind=kind,
-                this=this,
-                partition=partition,
-                expression=self._parse_compute_statistics(),
+        partition = self._parse_partition()
+        inner_expression: t.Optional[exp.Expression] = None
+        if self._match_texts(self.ANALYZE_EXPRESSION_PARSERS):
+            inner_expression = self.ANALYZE_EXPRESSION_PARSERS[self._prev.text.upper()](self)
+
+        properties = self._parse_properties()
+        return self.expression(
+            exp.Analyze,
+            kind=kind,
+            this=this,
+            partition=partition,
+            properties=properties,
+            expression=inner_expression,
+            options=options,
+        )
+
+    # https://spark.apache.org/docs/3.5.1/sql-ref-syntax-aux-analyze-table.html
+    def _parse_analyze_statistics(self, kind) -> exp.Statistics:
+        this = None
+        option = self._prev.text.upper() if self._match_text_seq("DELTA") else None
+        expressions = []
+
+        if not self._match_text_seq("STATISTICS"):
+            self.raise_error("Expecting token STATISTICS")
+
+        if self._match_text_seq("NOSCAN"):
+            this = "NOSCAN"
+        elif self._match(TokenType.FOR):
+            if self._match_text_seq("ALL", "COLUMNS"):
+                this = "FOR ALL COLUMNS"
+            if self._match_texts("COLUMNS"):
+                this = "FOR COLUMNS"
+                expressions = self._parse_csv(self._parse_column_reference)
+        elif self._match_text_seq("SAMPLE"):
+            sample = self._parse_number()
+            expressions = (
+                [self.expression(exp.Command, this="SAMPLE", expression=f" {sample} PERCENT")]
+                if self._match(TokenType.PERCENT)
+                else []
```

Comment:
@Why are you constructing a command here? This doesn't look right. We have a parser for the `SAMPLE` clause if you need to use that.

Reply:
We shouldn't hack this together using `Command`, that expression usually wraps around the whole input. You should parse this clause properly, or make sure to fallback into the `Command` case for the whole input (see other comment).